org.adempiere.base/src/org/compiere/util/DB.java

```diff
@@ -240,17 +245,30 @@ public static void updateMail()
 		no = DB.executeUpdate(sql.toString(), null);
 		if (log.isLoggable(Level.FINE)) log.fine("User #"+no);
 		//
+		FileOutputStream out = null;
 		try
 		{
 			env.setProperty("ADEMPIERE_MAIL_UPDATED", "Y");
-			FileOutputStream out = new FileOutputStream(envFile);
+			out = new FileOutputStream(envFile);
 			env.store(out, "");
-			out.flush();
-			out.close();

```

Comment:
the latest recommended pattern is try (FileOutputStream out = new FileOutputStream(envFile)) instead of close in finally clause.